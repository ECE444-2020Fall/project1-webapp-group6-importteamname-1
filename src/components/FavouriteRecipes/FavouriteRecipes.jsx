import React from "react";
import Button from '@material-ui/core/Button';

const FavouriteRecipes = () => {
  return <main id="mainContent">
    <div className="container">
      <div className="row justify-content-center mt-5 p-0">
        <h3>Favourite Recipes</h3>
      </div>
      <Button variant="contained" color="primary">
          Sample Button
      </Button>
    </div>
  </main>;
}
export default FavouriteRecipes;
