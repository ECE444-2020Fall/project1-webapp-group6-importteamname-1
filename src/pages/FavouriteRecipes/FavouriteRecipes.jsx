import React, { useState } from 'react';
import RecipeCard from "../../components/common/RecipeCard";
import { Link } from 'react-router-dom';
import CONSTANTS from '../../constants'

const FavouriteRecipes = () => {
  const [refresh, setRefresh] = useState(true);
  const [favouriteRecipes, setFavouriteRecipes] = useState([])

  if (refresh) {
    setRefresh(false)
    fetch(CONSTANTS.ENDPOINT.FAVOURITES_LIST, {
      credentials:'include'
    })
      .then(response => response.json())
      .then(response => setFavouriteRecipes(response.recipes));
  }

  let recipeSearchResult = null;

  if (favouriteRecipes.length != 0) {
    recipeSearchResult = <div>
      {favouriteRecipes.map(recipe => (
        <Link key={recipe.recipe_id} to={`recipe-search-results/${recipe.recipe_id}`}>
          <RecipeCard key={recipe.recipe_id}
            recipeId={recipe.recipe_id}
            recipeName={recipe.recipe_name}
            imageUrl={recipe.image_url}
            timeToCookInMinutes={recipe.time_to_cook_in_minutes}
            servings={recipe.servings}
            calories={recipe.calories}
            protein={recipe.protein}
            carbs={recipe.carbs}
            fat={recipe.fat} />
        </Link>
      ))}
    </div>;
  } else {
    recipeSearchResult = <p>No recipes favourited.</p>;
  }

  return (
    <main id="mainContent" className="container">
      <div className="row justify-content-center py-5">
        <h3>Favourite Recipes</h3>
      </div>
      {recipeSearchResult}
    </main>
  );
};

export default FavouriteRecipes;