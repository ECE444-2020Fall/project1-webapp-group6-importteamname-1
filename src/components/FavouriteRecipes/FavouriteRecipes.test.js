import React from 'react';
import FavouriteRecipes from './FavouriteRecipes';

import { configure, render } from 'enzyme';
import Adapter from 'enzyme-adapter-react-16';
import { shallow } from 'enzyme';
import ShallowRenderer from 'react-test-renderer/shallow';
configure({ adapter: new Adapter() });

describe('FavouriteRecipes', () => {
    it("renders FavouriteRecipes without crashing", () => {
        const FavouriteRecipesPage = shallow(<FavouriteRecipes />);
        expect(FavouriteRecipesPage).toMatchSnapshot();
    });

    it("renders <React.Fragment />", () => {
        const FavouriteRecipesPage = shallow(<FavouriteRecipes />);
        expect(FavouriteRecipesPage.find('h3').text()).toEqual("Favourite Recipes");
    });
});


  
