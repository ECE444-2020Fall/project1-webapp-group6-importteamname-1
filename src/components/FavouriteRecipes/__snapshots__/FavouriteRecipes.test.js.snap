// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`FavouriteRecipes renders FavouriteRecipes without crashing 1`] = `ShallowWrapper {}`;

exports[`FavouriteRecipes renders NavBar without crashing 1`] = `ShallowWrapper {}`;
